modules/sample_app/codesuite.tf

```diff
@@ -0,0 +1,452 @@
+    target_group_info {
+      name = "${var.app_name}-tgrp"
+    }
+  }
+
+  ec2_tag_set {
+    ec2_tag_filter {
+      key   = "Name"
+      type  = "KEY_AND_VALUE"
+      value = "ARC_App"
+    }
+  }
+
+  # trigger a rollback on deployment failure event
+  auto_rollback_configuration {
+    enabled = true
+    events = [
+      "DEPLOYMENT_FAILURE",
+    ]
+  }
+}
+
+
+# create a CodeDeploy application
+resource "aws_codedeploy_app" "app_region_2" {
+  provider = aws.alternative
+  name     = "ARC_App"
+}
+
+
+# create a deployment group
+resource "aws_codedeploy_deployment_group" "deployment_group_region_2" {
+  provider = aws.alternative
+
+  app_name              = aws_codedeploy_app.app_region_2.name
+  deployment_group_name = "ARC_Deployment_Group"
+  service_role_arn      = aws_iam_role.codedeploy_service.arn
+
+  deployment_config_name = "CodeDeployDefault.OneAtATime" # AWS defined deployment config
+
+  autoscaling_groups = [module.app_alternative.asg.name]
+
+  deployment_style {
+    deployment_option = "WITH_TRAFFIC_CONTROL"
+    deployment_type   = "IN_PLACE"
+  }
+
+
+  load_balancer_info {
+    target_group_info {
+      name = "${var.app_name}-tgrp"
+    }
+  }
+
+  ec2_tag_set {
+    ec2_tag_filter {
+      key   = "Name"
+      type  = "KEY_AND_VALUE"
+      value = "ARC_App"
+    }
+  }
+
+  # trigger a rollback on deployment failure event
+  auto_rollback_configuration {
+    enabled = true
+    events = [
+      "DEPLOYMENT_FAILURE",
+    ]
+  }
+}
+
+# Codepipeline role
+
+resource "aws_iam_role" "codepipeline_role" {
+  depends_on = [aws_s3_bucket.artifact_bucket_region_1, aws_s3_bucket.artifact_bucket_region_2]
+
+  assume_role_policy = <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Action": "sts:AssumeRole",
+      "Principal": {
+        "Service": "codepipeline.amazonaws.com"
+      },
+      "Effect": "Allow"
+    }
+  ]
+}
+EOF
+  path               = "/"
+}
+
+resource "aws_iam_policy" "codepipeline_policy" {
+  description = "Policy to allow codepipeline to execute"
+
+  depends_on = [aws_s3_bucket.artifact_bucket_region_1, aws_s3_bucket.artifact_bucket_region_2]
+
+  policy = <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Action": [
+        "s3:*"
+      ],
+      "Effect": "Allow",
+      "Resource": ["${aws_s3_bucket.artifact_bucket_region_1.arn}/*",
+                    "${aws_s3_bucket.artifact_bucket_region_1.arn}",
+                    "${aws_s3_bucket.artifact_bucket_region_2.arn}/*",
+                    "${aws_s3_bucket.artifact_bucket_region_2.arn}",
+                    "${aws_s3_bucket.app_source_code.arn}/*",
+                    "${aws_s3_bucket.app_source_code.arn}"]
+    },
+    {
+      "Action" : [
+        "iam:PassRole",
+        "codebuild:StartBuild",
+        "codebuild:BatchGetBuilds"
+      ],
+      "Effect": "Allow",
+      "Resource": "*"
+    },
+    {
+      "Effect" : "Allow",
+      "Action" : [
+        "kms:*"
```

Comment:
Same with this, please restrict this down to a single key or less permissive actions, or both

Reply:
this looks like it can be eliminated entirely. for context, this code was adapted and copied from: https://aws.amazon.com/blogs/networking-and-content-delivery/running-recovery-oriented-applications-with-amazon-route-53-application-recovery-controller-aws-ci-cd-tools-and-terraform/